src/Spice86.Core/Emulator/Devices/Video/AeonCard.cs

```diff
@@ -0,0 +1,836 @@
+    }
+
+    public void TickRetrace() {
+        // Inactive at tick time, but will become active once the code checks for it.
+        _crtStatusRegister = StatusRegisterRetraceInactive;
+    }
+
+    public void UpdateScreen() {
+        _gui?.UpdateScreen();
+    }
+
+    public event EventHandler? VideoModeChanged;
+
+    private void FillDispatchTable() {
+        _dispatchTable.Add(0x00, new Callback.Callback(0x00, SetVideoMode));
+        _dispatchTable.Add(0x01, new Callback.Callback(0x01, SetCursorType));
+        _dispatchTable.Add(0x02, new Callback.Callback(0x02, SetCursorPosition));
+        _dispatchTable.Add(0x03, new Callback.Callback(0x03, GetCursorPosition));
+        _dispatchTable.Add(0x05, new Callback.Callback(0x05, SelectActiveDisplayPage));
+        _dispatchTable.Add(0x06, new Callback.Callback(0x06, ScrollPageUp));
+        _dispatchTable.Add(0x07, new Callback.Callback(0x07, ScrollPageDown));
+        _dispatchTable.Add(0x08, new Callback.Callback(0x08, ReadCharacterAndAttributeAtCursor));
+        _dispatchTable.Add(0x09, new Callback.Callback(0x09, WriteCharacterAndAttributeAtCursor));
+        _dispatchTable.Add(0x0A, new Callback.Callback(0x0A, WriteCharacterAtCursor));
+        _dispatchTable.Add(0x0B, new Callback.Callback(0x0B, SetColorPaletteOrBackGroundColor));
+        _dispatchTable.Add(0x0E, new Callback.Callback(0x0E, WriteTextInTeletypeMode));
+        _dispatchTable.Add(0x0F, new Callback.Callback(0x0F, GetVideoMode));
+        _dispatchTable.Add(0x10, new Callback.Callback(0x10, GetSetPaletteRegisters));
+        _dispatchTable.Add(0x11, new Callback.Callback(0x11, CharacterGeneratorRoutine));
+        _dispatchTable.Add(0x12, new Callback.Callback(0x12, VideoSubsystemConfiguration));
+        _dispatchTable.Add(0x13, new Callback.Callback(0x13, WriteString));
+        _dispatchTable.Add(0x1A, new Callback.Callback(0x1A, VideoDisplayCombination));
+        _dispatchTable.Add(0x1B, new Callback.Callback(0x1B, GetFunctionalityInfo));
+    }
+
+    private void WriteString() {
+        uint address = new SegmentedAddress(_state.ES, _state.BP).ToPhysical();
+        StringBuilder res = new();
+        for (uint i = address; i < address + 1024; i++) {
+            byte characterByte = _memory.GetUint8(i);
+            if (characterByte == 0) {
+                break;
+            }
+
+            char character = Convert.ToChar(characterByte);
+            res.Append(character);
+        }
+
+        if (_loggerService.IsEnabled(LogEventLevel.Information)) {
+            _loggerService.Information("WRITE STRING: {0}", res.ToString());
+        }
+    }
+
+    private void SetVideoMode() {
+        SetVideoModeInternal((VideoModeId)_state.AL);
+        _bios.VideoMode = _state.AL;
+    }
+
+    private void GetFunctionalityInfo() {
+        ushort segment = _state.ES;
+        ushort offset = _state.DI;
+
+        uint address = MemoryUtils.ToPhysicalAddress(segment, offset);
```

Comment:
Would be nice to write a memory based structure to represent this instead to make it easier to read and also so that rewritten code can reuse it.
Example here: https://github.com/OpenRakis/Spice86/blob/master/src/Spice86.Core/Emulator/OperatingSystem/DosMemoryControlBlock.cs

Reply:
Done